testutil/integration/ibc/chain/chain.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package chain
+
+import (
+	"time"
+
+	tmtypes "github.com/cometbft/cometbft/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+	"github.com/cosmos/ibc-go/v7/testing/simapp"
+)
+
+// IBCChain defines the required methods needed for a testing IBC chain that complies
+// with the ibctesting chain struct.
+type Chain interface {
+	// GetContext returns the current context for the application.
+	GetContext() sdktypes.Context
+	// GetSimApp returns the SimApp to allow usage ofnon-interface fields.
+	GetSimApp() *simapp.SimApp
```

Comment:
how come we need to redefine this, I would have expected there is a corresponding interface on the IBC-Go repo which we could import?

Reply:
There's not 🤦‍♂️ that would have ideal because there are parts of the implementation side that I don't like. Everything's tight together and that's what I'm trying to get out of